scripts/db/src/get-user.ts

```diff
@@ -50,48 +64,100 @@ async function getByEmail(email: string, callback: DbScriptCallback) {
       POSTGRES_HOST,
       POSTGRES_DATABASE,
       POSTGRES_PORT,
+      MONGO_URI,
+      MONGO_DB_NAME,
     } = (configuration as unknown) as DbConfiguration
 
-    /** Declare connection info */
-    const connectionInfo: ConnectionConfig = {
-      user: POSTGRES_USERNAME,
-      password: POSTGRES_PASSWORD,
-      host: POSTGRES_HOST,
-      database: POSTGRES_DATABASE,
-      port: POSTGRES_PORT ? parseInt(POSTGRES_PORT) : 5432,
-      ssl: TEMPLATE_DATA.pgShouldSsl,
-    }
+    /**
+     * Logic in this function tries to match:
+     * EAForum/packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx
+     */
+    async function getForumUser(): Promise<CallbackUser | null> {
+      // Connect to the Forum DB
+      const mongoClient: MongoClient = new MongoClient(MONGO_URI)
+      await mongoClient.connect()
+
+      // Query the users collection for someone with our email
+      const forumUser = await mongoClient
+        .db(MONGO_DB_NAME)
+        .collection<ForumUser>('users')
+        .findOne({ 'emails.address': email })
+
+      await mongoClient.close()
+
+      if (!forumUser) {
+        return null
+      }
+
+      // Which email did we find?
+      const emailInfo = forumUser.emails.find((e) => e.address === email)
+      if (!emailInfo) {
+        // This should never happen, as they were returned by mongo because that
+        // field matched
+        throw new Error(
+          `User found by email ${email}, does not have that email (should never happen)`
+        )
+      }
 
-    /** Construct a postgres client and connect to the server */
-    const client: Client = new Client(connectionInfo)
-    await client.connect()
-
-    /** Get the person based on their email, joining on the password table */
-    const query = `
-      select
-        person.id, email, first_name, last_name, password
-      from people.person
-      join auth.password on password.person_id = person.id
-      where person.email = $1
-    `
-    const result = await client.query<PersonResult>(query, [email])
-
-    /** Close the connection */
-    await client.end()
-
-    if (result.rows.length === 0) {
-      return callback(null)
+      return {
+        id: forumUser._id,
+        username: forumUser.displayName,
```

Comment:
I think there was some discussion on Slack about whether to use `username` or `nickname`. From the docs, it looks like `username` is unique, whereas nickname is presumably more like a display name. Thinking about it very quickly, I think it makes sense for this to be a unique field, but that then makes me wonder if the DB field is actually `displayName` or whether it should actually be `username`...

Reply:
I think nickname is more likely to collide with, say, google in a bad way. But on the other hand, displayName, is unfortunately not guaranteed to be unique.

Reply:
Going back to nickname.